src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops.cpp

```diff
@@ -81,6 +100,10 @@ bool PerformanceImplicitGemmForwardV4R4Xdlops::SetNextValue()
     {
         // list performance parameters in reverse order, in order for tuning to iterate over the
         // range in normal order
+#if CK_ADD_VECTOR_LOAD_GEMMN_TUNE_PARAM
+        if(!NextTwoPower<1, 8>(GemmBThreadDataPerRead_GemmN))
```

Comment:
This increases tuning time 4 times. Is it acceptable?

Reply:
@atamazov , thanks for your comments.
This tune param is only valid for fp16's vectorizable case, also there's more restriction inside of "PerformanceImplicitGemmForwardV4R4Xdlops::CalculateGemmBBlockCopyPerformancePara", pls have a reference.

normally, for fp16's vectorizable cases, tuning time is 2 times of original time.

Reply:
Is it so that this implementation locks the value of this new param at some default value for all cases except "fp16 vectorizable"? And this is done in `PerformanceImplicitGemmForwardV4R4Xdlops::CalculateGemmBBlockCopyPerformancePara()`?

If yes, them why this is done there but not right here?

Reply:
the reason is that inside of 'SetNextValue', 'ConvolutionContext' is not accessible, so there's no way we can do seperate diverse for fp16.

Reply:
It seems like missing access to `ConvolutionContext` enforces us to use weird designs. However `ComputedContainer` is intended to ease the code (and the life of Solvers' developers).

We can make `ConvolutionContext` accessible in `SetNextValue()`.

What do you think?

/cc @shurale-nkn @asroy @ce1adon @zjing14 @daniellowell @DrizztDoUrden

Reply:
@atamazov , I agree, and adding convolutionContext into setNextValue would ease solver's tunning implementation, however this change would be framwork-level impacted. so I suggest finish this pr first and then open another pr to fix it, what do you think?

Reply:
@shurale-nkn @asroy @ce1adon @zjing14 @daniellowell @DrizztDoUrden
Could you pls comments on idea of making ConvolutionContext accessible in SetNextValue()? 
and since its impacts framework change, how we can fulfill it?

Reply:
Obviously there is no direct impact on the existing code. It just _opens ways_ to write new code better, and to refactor existing code to make it cleaner.

Reply:
> It seems like missing access to `ConvolutionContext` enforces us to use weird designs. However `ComputedContainer` is intended to ease the code (and the life of Solvers' developers).
> 
> We can make `ConvolutionContext` accessible in `SetNextValue()`.
> 
> What do you think?

That's nice. Having accessing to ```ConvolutionContext``` anywhere is always good.

Reply:
@atamazov, @asroy  I've tried to make ConvolutionContext accessible in SetNextValue(), however other solver's SetNextValue() does not use ConvolutionContext, so there would be warning while compile. so code change can not pass the CI.
So I change to add a local value of m_isFp16 inside of PerformanceImplicitGemmForwardV4R4Xdlops, since it's only needed inside of this solver currently. 
Pls review the change.